chat.go

```diff
@@ -55,7 +56,26 @@ type ChatCompletionRequest struct {
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
 	Functions        []*FunctionDefine       `json:"functions,omitempty"`
-	FunctionCall     string                  `json:"function_call,omitempty"`
+	// FunctionCall can either be a string ("none" or "auto") or an instance of ChatCompletionRequestFunctionCall
+	FunctionCall any `json:"function_call,omitempty"`
+}
+
+func checkIsChatCompletionFunctionCall(value any) bool {
```

Comment:
Should this maybe be named something like `validateChatCompletionFunctionCall`? The current name makes me think a return of `true` would mean it is a function call, but `true` means it is a valid value (even if there is no function call involved).